pkg/proxy/healthcheck/healthcheck.go

```diff
@@ -339,7 +339,7 @@ func (h healthzHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 
 	resp.Header().Set("Content-Type", "application/json")
 	if !lastUpdated.IsZero() && currentTime.After(lastUpdated.Add(h.hs.healthTimeout)) {
-		resp.WriteHeader(http.StatusServiceUnavailable)
+		http.Error(resp, "", http.StatusServiceUnavailable)
```

Comment:
same here, we shouldn't send json after resetting to text/plain

Reply:
will defer to @kubernetes/sig-network-api-reviews as well, or we can restore current behavior and work through a change here in a follow-up.

Reply:
I've updated the PR to keep the current behavior here.